features/step_definitions/kata.rb

```diff
@@ -227,3 +227,78 @@
     end
   end
 end
+
+Given /^Verify catalog source existence$/ do
+  project_name = 'openshift-marketplace'
+  catalog_source_name = 'redhat-operators'
+  step %Q/I switch to cluster admin pseudo user/
+  project(project_name)
+  if !catalog_source(catalog_source_name).exists?
+    logger.warn("Missing #{catalog_source_name} catalog source")
+    logger.info("Creating #{catalog_source_name}")
+    step %Q/I create "redhat-operators" catalogsource for testing/
+    sleep(120)
```

Comment:
hardcoded wait is discouraged.

Reply:
Do we have a wrapped sleep function o something?

Reply:
@pruan-rht Can you point us to the coding standards for openshift QE test automation?  We know these aren't perfect.  If it is not violating some standard that is required we would like to make iterative improvements rather than making it perfect the first time.

Reply:
@cpmeadors @valtair1891 https://github.com/openshift/verification-tests/blob/master/doc/coding_guide.adoc

Reply:
@cpmeadors , @valtair1891 , we usually wait for some deployment, pod, service or route to become available and have steps for the purpose. The reason is that sometimes operations take a very short time, in other clusters 3 or 5 times more time or during other load.

To avoid making tests super slow and always play catch-up with wait times, we try to avoid hardcoded sleep. It is harder, I know but the other option is very bad at runtime.